fcsparser/api.py

```diff
@@ -206,16 +207,23 @@ def _extract_text_dict(self, raw_text):
                 raise ParserFeatureNotImplementedError(msg)
 
         # Below 1:-1 used to remove first and last characters which should be reserved for delimiter
-        # Also, the delimiter is "quoted" by being repeated (two consecutive delimiters)
-        delimiter_split_arr = [x.split(delimiter) for x in raw_text[1:-1].split(delimiter * 2)]
-
-        raw_text_segments = delimiter_split_arr[0]
-        for partial_segment in delimiter_split_arr[1:]:
-            raw_text_segments[-1] += (delimiter + partial_segment[0])
-            raw_text_segments.extend(partial_segment[1:])
+        raw_text = raw_text[1:-1]
+
+        ed = re.escape(delimiter)  # escaped for regex, not FCS format
+        # The delimiter is escaped by being repeated (two consecutive delimiters). This regex finds
+        # odd-length repetitions of the delimiter. Since each pair is escaped, only the last
+        # delimiter in odd-length repetitions truly delimits TEXT keys and values.
+        regex = "(?<!" + ed + ")(" + ed * 2 + ")*" + ed + "(?!" + ed + ")"
```

Comment:
Let me think about this a bit tonight to see if I can come up with anything different (the previous version you had is actually easier to read)

Reply:
For what it's worth, I had to go through a few different iterations of the code to get to the previous version. Maybe I revert to that, and add some better comments?

Reply:
I apologize I forgot to look at it -- but yes that sounds good

Reply:
Sorry for making you go through the extra trouble!

Reply:
No problem - it got me to finally learn how lookaheads work.